quinn-proto/src/connection.rs

```diff
@@ -529,13 +529,19 @@ impl Connection {
         match timer {
             Timer::Close => {
                 self.state = State::Drained;
+                if self.app_closed {
```

Comment:
Similarly to `init_cid`, I think `app_closed` is the `Endpoint`'s' concern, and could be removed from `Connection` entirely; instead, the `Connection` oculd notify the `Endpoint` whenever it becomes `Drained`, and the `Endpoint` would dispose of the `Connection` when it has both become `Drained` and been explicitly closed by the user.

Reply:
Done.